assets/javascript/game.js

```diff
@@ -0,0 +1,335 @@
+$(document).ready(function() {
+  // Launch the initial game
+  wordGame.pickRandomWord();
+
+  $("#game-name").text(wordGame.getName());
+  $("#mystery-word").text(wordGame.getPartialWord());
+  $("#guesses-remaining").text(wordGame.getGuessesRemaining());
+  $("#player-wins").text(wordGame.getWins());
+  giveUserFeedback(wordGame.getInstructions());
+
+  $(document).keyup(function(event) {
+    // DEBUG
+    // alert("You pressed the " + event.key + " key!");
+   
+    // Don't allow further input if game is over.
+    if (wordGame.getGameOver()) { 
+      return;
+    }
+    // Only accept valid user input (*i.e.*, alphabetic letters)
+    else if (event.key.length === 1 && 
+            (event.key.charCodeAt(0) >= 97 && event.key.charCodeAt(0) <= 122)) {
+      // DEBUG
+      // console.log(event.key.charCodeAt(0));
+
+      (event.key.charCodeAt(0));
+      if(wordGame.isLtrInGuesses(event.key)) {
+        giveUserFeedback("Letter already guessed.");
+        return;
+      }
+
+      if (wordGame.isLtrInWord(event.key)) {
+        giveUserFeedback("Great job! Keep going.");
+      } 
+      else {
+        giveUserFeedback("Oops! Bad guess. Try again.");
+      }
+
+      wordGame.decrementGuesses();   
+      wordGame.addGuessToList(event.key);
+
+      $("#mystery-word").text(wordGame.getPartialWord());
+      $("#letters-guessed").text(wordGame.getLettersGuessed());
+      $("#guesses-remaining").text(wordGame.getGuessesRemaining());
+
+      // Check for a win!
+      if (!(wordGame.getPartialWord().includes("_"))) {
+        wordGame.setGameOver();
+        wordGame.incrementWins();
+
+        var randomObj = wordGame.getRandomObj();
+
+        $("#winning-prize-container img").attr({
+          src: randomObj.img,
+          alt: randomObj.word
+        });
+        $("#winning-prize-container figcaption").text(randomObj.temperament.toUpperCase());
+
+        giveUserFeedback("WE HAVE A WINNER!", "win");
+      }
+      // Check for loss.
+      else if (wordGame.getGuessesRemaining() < 1) {
+        wordGame.setGameOver();
+        wordGame.incrementLosses();
+    
+        giveUserFeedback("You lost this round. Better luck next time.", "loss");
+      }
+
+      var gamesPlayed = wordGame.getWins() + wordGame.getLosses();
+
+      $("#player-wins").text(wordGame.getWins() + " out of " + gamesPlayed + " game(s)" );
+
+      // Check if game is over.
+      if (wordGame.getGameOver()) {
+        $("#play-again").append("<button>");
+        $("#play-again > button").attr({
+          type: "button",
+          class: "btn btn-outline-light"
+        });
+        $("#play-again > button").text("Play Again!");
+      }
+
+    }
+    // Notify user input was invalid. 
+    else {
+      giveUserFeedback("Oops! Invalid character pressed. Try again.");
+    }
+  });
+
+  // If user requests another game, reset playing field.
+  $(document).on("click", function(event) {
+    if (event.target.type === "button") {
```

Comment:
Also I would extract this to a named function to make it clear what's going on even without comments:

```javascript
function resetGame(event) {
  wordGame.resetGameState();
  // ...
}

$('button').on('click', resetGame);
```

Nothing against comments though!